Error saving cipher

[2019-02-04 22:45:47][bitwarden_rs::db::models::user][WARN] Failed to update revision for xxxxxxxx@xxxx.fr
[2019-02-04 22:45:47][bitwarden_rs::error][ERROR] Error saving cipher.
[CAUSE] DatabaseError(
    __Unknown,
    "database is locked"
)

--- a/src/api/core/ciphers.rs
+++ b/src/api/core/ciphers.rs
@@ -853,7 +853,6 @@ fn move_cipher_selected(data: JsonUpcase<MoveCipherData>, headers: Headers, conn
 
         // Move cipher
         cipher.move_to_folder(data.FolderId.clone(), &user_uuid, &conn)?;
-        cipher.save(&conn)?;
 
         nt.send_cipher_update(
             UpdateType::CipherUpdate,

--- a/src/api/core/ciphers.rs
+++ b/src/api/core/ciphers.rs
@@ -857,7 +857,7 @@ fn move_cipher_selected(data: JsonUpcase<MoveCipherData>, headers: Headers, conn
         nt.send_cipher_update(
             UpdateType::CipherUpdate,
             &cipher,
-            &User::update_uuid_revision(&user_uuid, &conn),
+            &vec![user_uuid.clone()]
         );
     }
 

--- a/src/api/core/ciphers.rs
+++ b/src/api/core/ciphers.rs
@@ -857,7 +857,7 @@ fn move_cipher_selected(data: JsonUpcase<MoveCipherData>, headers: Headers, conn
         nt.send_cipher_update(
             UpdateType::CipherUpdate,
             &cipher,
-            &vec![user_uuid.clone()]
+            &[user_uuid.clone()]
         );
     }
 

--- a/src/api/core/ciphers.rs
+++ b/src/api/core/ciphers.rs
@@ -842,7 +842,7 @@ fn move_cipher_selected(data: JsonUpcase<MoveCipherData>, headers: Headers, conn
     }
 
     for uuid in data.Ids {
-        let mut cipher = match Cipher::find_by_uuid(&uuid, &conn) {
+        let cipher = match Cipher::find_by_uuid(&uuid, &conn) {
             Some(cipher) => cipher,
             None => err!("Cipher doesn't exist"),
         };

--- a/src/db/models/user.rs
+++ b/src/db/models/user.rs
@@ -184,10 +184,12 @@ impl User {
 
     pub fn update_revision(&mut self, conn: &DbConn) -> EmptyResult {
         self.updated_at = Utc::now().naive_utc();
-        diesel::update(users::table.filter(users::uuid.eq(&self.uuid)))
-            .set(users::updated_at.eq(&self.updated_at))
-            .execute(&**conn)
-            .map_res("Error updating user revision")
+        crate::util::retry( || {
+            diesel::update(users::table.filter(users::uuid.eq(&self.uuid)))
+                .set(users::updated_at.eq(&self.updated_at))
+                .execute(&**conn)
+        }, 10)
+        .map_res("Error updating user revision")
     }
 
     pub fn find_by_mail(mail: &str, conn: &DbConn) -> Option<Self> {

